backport-2.0: compactor: configure via cluster settings

tbg opened the pull request:
Backport 1/1 commits from #25397.

/cc @cockroachdb/release

---

The compactor can cause instability or degrade performance, and it does
so in an intransparent way, partially because its compactions are
low-priority and are thus regularly pending for upwards of ten minutes,
without being able to tell at which point work is being done. It's
important to be able to change its behavior, and in particular to
disable it, whether that's because of a known problem or a suspected
one. This wasn't possible at all previously. Now it can be achieved by
setting `compactor.threshold_bytes` and `max_record_age` to a very high
and low value, respectively.

Release note (general change): Introduced cluster settings to configure
the compactor.

tbg committed:
compactor: configure via cluster settings

The compactor can cause instability or degrade performance, and it does
so in an intransparent way, partially because its compactions are
low-priority and are thus regularly pending for upwards of ten minutes,
without being able to tell at which point work is being done. It's
important to be able to change its behavior, and in particular to
disable it, whether that's because of a known problem or a suspected
one. This wasn't possible at all previously. Now it can be achieved by
setting `compactor.enabled` to `false`; the advanced settings have
been marked as hidden.

Release note (general change): Introduced cluster settings to configure
the compactor.

a-robinson reviewed (approved):
LGTM, but you may want to check out https://github.com/cockroachdb/cockroach/issues/25459 before putting this onto release-2.0

tbg committed:
storage: fix compactor-disabled test

The test was queuing suggestions worthy of a compaction before disabling
the compactor.

Also fixes a bug in the compactor that would over-increment the bytes
skipped metric whenever multiple suggestions were aggregated and
skipped.

Release note: None

tschottdorf commented:
I cherry-picked 2020272ae on top of this.

tschottdorf commented:
bors r=a-robinson
